### ISSUE TITLE
Add Capacitance Matrix Method (CMM) for DST Helmholtz solver on irregular domains

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#72]
## Description

`finitevolX` has a `HelmholtzDST2D` spectral solver using Discrete Sine Transforms for rectangular domains. However, it has **no support for irregular (masked) domains** in the spectral solver path. The reference louity repos solve this with the **Capacitance Matrix Method (CMM)** — a technique that extends DST-based spectral solvers to masked domains by:

1. First solving on the full rectangle (ignoring the mask)
2. Computing correction source terms at irregular boundary points using a precomputed capacitance matrix (inverse of a Green's function matrix)
3. Solving a second time with the corrected source to enforce homogeneous Dirichlet at mask boundaries

This is much faster than multigrid for moderate irregular boundary point counts and maintains the O(N log N) spectral solve cost.

### Reference Implementations

**From [`louity/MQGeometry/helmholtz.py`](https://github.com/louity/MQGeometry/blob/1511b1a9bbc4b1213017f59305e2ce466b308169/helmholtz.py#L37-L72):**

```python
def compute_capacitance_matrices(helmholtz_dst, bound_xids, bound_yids):
    nl = helmholtz_dst.shape[-3]
    M = bound_xids.shape[0]

    # compute G matrices (Green's function at boundary points)
    G_matrices = torch.zeros((nl, M, M), dtype=torch.float64, device='cpu')
    rhs = torch.zeros(helmholtz_dst.shape[-3:], ...)
    for m in range(M):
        rhs.fill_(0)
        rhs[..., bound_xids[m], bound_yids[m]] = 1
        sol = dstI2D(dstI2D(rhs) / helmholtz_dst)
        G_matrices[:,m] = sol[..., bound_xids, bound_yids].cpu()

    # invert G matrices to get capacitance matrices
    capacitance_matrices = torch.zeros_like(G_matrices)
    for l in range(nl):
        capacitance_matrices[l] = torch.linalg.inv(G_matrices[l])
    return capacitance_matrices

def solve_helmholtz_dst_cmm(rhs, helmholtz_dst, cap_matrices,
                            bound_xids, bound_yids, mask):
    sol_rect = dstI2D(dstI2D(rhs) / helmholtz_dst)
    alphas = torch.einsum('...ij,...j->...i', cap_matrices,
                          -sol_rect[..., bound_xids, bound_yids])
    rhs_2 = rhs.clone()
    rhs_2[..., bound_xids, bound_yids] = alphas
    sol = dstI2D(dstI2D(rhs_2) / helmholtz_dst)
    return F.pad(sol, (1,1,1,1)) * mask
```

**From [`louity/qgsw-pytorch/src/helmholtz.py`](https://github.com/louity/qgsw-pytorch/blob/673c8e60288baaf1e426dfc3ddd4266380f781cf/src/helmholtz.py):**
Same pattern with `solve_helmholtz_dstI_cmm` and `compute_capacitance_matrices`.

## Proposed API

```python
def compute_capacitance_matrices(
    helmholtz_dst: Array,       # spectral Helmholtz operator eigenvalues
    bound_xids: Array,          # x-indices of irregular boundary points
    bound_yids: Array,          # y-indices of irregular boundary points
) -> Array:
    """Precompute capacitance matrices for CMM on irregular boundaries.

    Returns shape (nl, M, M) where M = number of boundary points.
    """

def solve_helmholtz_dst_cmm(
    rhs: Array,
    helmholtz_dst: Array,
    cap_matrices: Array,
    bound_xids: Array,
    bound_yids: Array,
    mask: Array,
) -> Array:
    """Solve Helmholtz equation using DST + Capacitance Matrix Method.

    Works on irregular (masked) domains by correcting the rectangular
    DST solution to enforce boundary conditions at mask boundaries.
    """
```

Should integrate with the existing `HelmholtzDST2D` solver as an optional extension when a mask is provided.

## Acceptance Criteria

- [ ] `compute_capacitance_matrices` function
- [ ] `solve_helmholtz_dst_cmm` function
- [ ] Integration with existing `HelmholtzDST2D` (auto-detect masked domain)
- [ ] Extract irregular boundary indices from `ArakawaCGridMask`
- [ ] Tests: round-trip on rectangular domain matches pure DST
- [ ] Tests: solve on circular mask, verify solution satisfies BCs
- [ ] Tests: verify solution matches multigrid solver (#71) on same problem
- [ ] Benchmark: show CMM is faster than multigrid for small boundary point counts
- [ ] JAX-compatible

## Priority
**Medium-High** — extends the fast spectral solver to irregular geometries without multigrid overhead.

---

_References:_
- _[louity/MQGeometry/helmholtz.py](https://github.com/louity/MQGeometry/blob/1511b1a9bbc4b1213017f59305e2ce466b308169/helmholtz.py#L37-L72)_
- _[louity/qgsw-pytorch/src/helmholtz.py](https://github.com/louity/qgsw-pytorch/blob/673c8e60288baaf1e426dfc3ddd4266380f781cf/src/helmholtz.py)_
